src/ChangesReporting/Output/GitHubOutputFormatter.php

```diff
@@ -0,0 +1,155 @@
+                ? $systemError->getAbsoluteFilePath()
+                : $systemError->getRelativeFilePath();
+
+            $line = $systemError->getLine();
+            $message = trim($systemError->getRectorShortClass() . PHP_EOL . $systemError->getMessage());
+
+            $this->reportErrorAnnotation($message, [
+                'file' => $filePath,
+                'line' => $line,
+            ]);
+        }
+    }
+
+    private function reportFileDiffs(ProcessResult $processResult, Configuration $configuration): void
+    {
+
+        $fileDiffs = $processResult->getFileDiffs();
+        ksort($fileDiffs);
+
+        foreach ($fileDiffs as $fileDiff) {
+            $filePath = $configuration->isReportingWithRealPath()
+                ? $fileDiff->getAbsoluteFilePath()
+                : $fileDiff->getRelativeFilePath();
+
+            $line = $fileDiff->getFirstLineNumber();
+            $endLine = $fileDiff->getLastLineNumber();
+
+            $message = trim(
+                implode(' / ', $fileDiff->getRectorShortClasses())
+            ) . PHP_EOL . PHP_EOL . $fileDiff->getDiff();
+
+            $this->reportErrorAnnotation($message, [
+                'file' => $filePath,
+                'line' => $line,
+                'endLine' => $endLine,
+            ]);
+        }
+    }
+
+    /**
+     * @param AnnotationProperties $annotationProperties
+     */
+    private function reportErrorAnnotation(string $message, array $annotationProperties): void
+    {
+        $properties = $this->sanitizeAnnotationProperties($annotationProperties);
+
+        $command = sprintf('::error %s::%s', $properties, $message);
+
+        // Sanitize command
+        $command = str_replace(['%', "\r", "\n"], ['%25', '%0D', '%0A'], $command);
+
+        echo $command . PHP_EOL;
+    }
+
+    /**
+     * @param AnnotationProperties $annotationProperties
+     */
+    private function sanitizeAnnotationProperties(array $annotationProperties): string
+    {
+        if (! isset($annotationProperties['line']) || ! $annotationProperties['line']) {
+            $annotationProperties['line'] = 0;
+        }
+
+        // This is a workaround for buggy endLine. See https://github.com/orgs/community/discussions/129899
+        // TODO: Should be removed once github will have fixed it issue.
+        unset($annotationProperties['endLine']);
+
+        $nonNullProperties = array_filter($annotationProperties, static fn ($value) => $value !== null);
+
+        $sanitizedProperties = array_map(
+            fn ($key, $value) => sprintf('%s=%s', $key, $this->sanitizeAnnotationProperty($value)),
+            array_keys($nonNullProperties),
+            $nonNullProperties
+        );
+
+        return implode(',', $sanitizedProperties);
+    }
+
+    /**
+     * @param string|int|null $value
+     * @return string
+     */
+    private function sanitizeAnnotationProperty($value)
+    {
+        if (! $value) {
+            return '';
+        }
```

Comment:
how about "0" string value? this should only check empty string imo, or just remove the check

Reply:
Totally right!